langchain/callbacks/streamlit.py

```diff
@@ -10,9 +10,10 @@
 class StreamlitCallbackHandler(BaseCallbackHandler):
     """Callback Handler that logs to streamlit."""
 
-    def __init__(self) -> None:
+    def __init__(self, container=st.container()) -> None:
```

Comment:
just checking, is output of st.container() mutable? if so we probably don't want to set that as default value

Reply:
I could be missing a potential use case, but right now I can't think of any situations where it needs to be mutable since you'd expect to read the output from the same expected region.
If the concern is to not introduce any unwanted side effects, we could maybe just use kwargs? something like this:

```
    def __init__(self, **kwargs) -> None:
        self.tokens_area = st.empty()
        self.tokens_stream = ""
        self.container = kwargs.get(container)
...
    def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
        """Run on new LLM token. Only available when streaming is enabled."""
        self.tokens_stream += token
        if self.container:
            with self.container:
                self.tokens_area.write(self.tokens_stream)
        else:
            self.tokens_area.write(self.tokens_stream)
```

Reply:
the recent commit b3c49e94a0ada8a1dacc5116611abcde0560846c using self.st actually fixed the origin issue behind this PR. I'll pull it back.